### PR TITLE
v2v Throttling moved to backend

### PR DESCRIPTION
**Pull request by jameswnl**
Throttling and some steps are moved to backend

depends on https://github.com/ManageIQ/manageiq/pull/18415/

### CODE REVIEW

**Comment by jameswnl** [comment]
@fdupont-redhat please help review and check the corresponding `manageiq` PR

**Comment by jameswnl** [comment]
@miq-bot remove_label wip
@miq-bot add_labels enhancement, tranformation

**Comment by jameswnl** [comment]
@miq-bot add_labels transformation

File: content/automate/ManageIQ/Transformation/StateMachines/VMTransformation.class/transformationcleanup.yaml [hunk 1]
```diff
@@ -1,37 +0,0 @@
----
```
Comment (fdupont-redhat) [inline content/automate/ManageIQ/Transformation/StateMachines/VMTransformation.class/transformationcleanup.yaml]: @jameswnl why do you remove this state machine ? It's called when migration fails to ensure that we return to initial state. But it's not always done...

Comment (jameswnl) [reply content/automate/ManageIQ/Transformation/StateMachines/VMTransformation.class/transformationcleanup.yaml]: Good question, please review [here](https://github.com/ManageIQ/manageiq/pull/18415/files#diff-51b91017cbf3d1648a52ca0482a63650R58)

Comment (fdupont-redhat) [reply content/automate/ManageIQ/Transformation/StateMachines/VMTransformation.class/transformationcleanup.yaml]: Reviewed, but I don't see where cancelation happens. Could you point to the methods that effectively kill virt-v2v and power the source VM on when the migration fails, not only on cancellation ?

Comment (jameswnl) [reply content/automate/ManageIQ/Transformation/StateMachines/VMTransformation.class/transformationcleanup.yaml]: How was this cleanup invoked previously?

**Comment by jameswnl** [comment]
@bzwei can you help take a look as well?

**Comment by jameswnl** [comment]
@miq-bot assign @gmcculloug

**Comment by gmcculloug** [comment]
@jameswnl As an enhancement I think this is `hammer/no`, but would ask that you set the proper label.

File: content/automate/ManageIQ/Transformation/Common.class/__methods__/vmchecktransformed.rb [hunk 2]
```diff
@@ -30,7 +30,7 @@ def update_total_percentage
           end
 
           def main
-            @task.get_conversion_state
+            # @task.get_conversion_state is now done from InfraConversionJob
```
Comment (gmcculloug) [inline content/automate/ManageIQ/Transformation/Common.class/__methods__/vmchecktransformed.rb]: @jameswnl Minor, but is there any reason to maintain this commented line?  I do not think it adds much value to retain it.

**Comment by fdupont-redhat** [comment]
> @jameswnl As an enhancement I think this is `hammer/no`, but would ask that you set the proper label.

@gmcculloug we target Hammer-3, which is planned for second half on April, IIRC. Or I'm lost in versions numbering...

**Comment by simaishi** [comment]
Hammer backport details:

```
$ git log -1
commit 439d0de4be15acc18caab7a4dc6e36b6d7906d4c
Author: Greg McCullough <gmccullo@redhat.com>
Date:   Mon Mar 4 17:39:54 2019 -0500

    Merge pull request #504 from jameswnl/throttle
    
    v2v Throttling moved to backend
    
    (cherry picked from commit a8f5cc3c4833c62791e329c793a1236e0b89df78)
    
    https://bugzilla.redhat.com/show_bug.cgi?id=1686877
```

**Pull request merged**